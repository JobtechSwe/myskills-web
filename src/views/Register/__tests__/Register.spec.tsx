// /* eslint-disable */
// import * as React from 'react'
// import Register, { GET_CONSENT_ID } from '../Register'

// import { render } from '../../../utils/test-utils'
// import { fireEvent, wait, waitForElement } from 'react-testing-library'

// const consent = {
//   id: 'db993c65-0673-454e-b951-bcb8d274f184',
//   expires: '1552561115',
//   url: '123',
//   __typename: 'Consent',
// }

// const getConsentMock = [
//   {
//     request: {
//       query: GET_CONSENT_ID,
//     },
//     result: {
//       data: {
//         consent,
//       },
//     },
//   },
// ]

// jest.mock('../Consent', () => () => (
//   <p>mydata://register/db993c65-0673-454e-b951-bcb8d274f184</p>
// ))

// describe('views/Register', () => {
//   it('renders without errors', async () => {
//     const { container } = render(<Register />, [])

//     await wait()

//     expect(container).toMatchSnapshot()
//   })

//   it('should render loading state initially', async () => {
//     const { getByText } = render(<Register />, getConsentMock)

//     fireEvent.click(getByText(/register/i))

//     expect(getByText(/loading.../i)).toBeInTheDocument()
//   })

//   it.skip('should render errors', async () => {
//     const errorMock = [
//       {
//         request: {
//           query: GET_CONSENT_ID,
//         },
//         result: {
//           error: new Error('Error while trying to request consentId'),
//         },
//       },
//     ]
//     const { getByText } = render(<Register />, errorMock)

//     fireEvent.click(getByText(/login/i))

//     await wait()

//     expect(getByText(/thatâ€™s an error./i)).toBeInTheDocument()
//   })

//   it('should render consentId after Register is clicked', async () => {
//     const { getByText } = render(<Register />, getConsentMock)

//     fireEvent.click(getByText(/register/i))

//     await waitForElement(() => getByText(`mydata://register/${consent.id}`))

//     expect(getByText(`mydata://register/${consent.id}`)).toBeInTheDocument()
//   })
// })
