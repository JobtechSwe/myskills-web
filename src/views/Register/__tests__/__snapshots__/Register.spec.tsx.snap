// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Register renders without errors 1`] = `
<div>
  <div
    class="css-128i4b2"
  >
    <button
      class="css-1qzvtil"
      font-family="default"
      font-size="medium"
    >
      Register
    </button>
  </div>
</div>
`;
